Gemfile

```diff
@@ -113,6 +113,10 @@ gem 'ofn-qz', github: 'openfoodfoundation/ofn-qz', branch: 'ofn-rails-4'
 
 gem 'good_migrations'
 
+gem "rack-cors"
```

Comment:
I'd prefer a different solution here, if there is one.

The problem was that loading StoryBook in dev was throwing CORS errors, right? I think we can probably solve that without adding a new gem.

Reply:
> The problem was that loading StoryBook in dev was throwing CORS errors, right?

Exactly. You can see the error in this comment: https://github.com/openfoodfoundation/openfoodnetwork/pull/7284#issuecomment-811356981

Just to understand: what is the main problem by adding some gems? Are you concerned about the systematic increase in the number of dependencies, which can be annoying at some times?

Reply:
> Are you concerned about the systematic increase in the number of dependencies, which can be annoying at some times?

Yeah, we should try to keep dependencies to a minimum if possible :+1:

Reply:
I think I have reached the limits of my skills with rails: I can use a `gem`, but replace it with our code seems to be (at the moment) too difficult for me. 
I'd really appreciate if you could help me on this one. 🙏

Reply:
(BTW, I can understand the concern about having an eye on the number of dependencies. On the other hand, I also concerned about the NIH syndrome, and I think each dependencies should be looked by itself and without being too categorical.)

Reply:
Matt is not suggesting to add a complex solution to our codebase that a gem could do better. He suggests that we try to understand why our codebase has an issue with iframes and then the solution may be very simple. A bit of OFN history knowledge helps here.

We started restricting iframes when we offered embedded shopfronts. They never worked properly and the feature is deprecated now. But the restriction is still there:
https://github.com/openfoodfoundation/openfoodnetwork/blob/2128675e955cc1f1cbae875bbac47dac0e8aafff/app/controllers/application_controller.rb#L93-L102

Maybe we can add a rule there to allow requests for Storybook.

Reply:
It's not exactly related to iframe, but more about CORS policy.
But anyway, will look to `openfoodnetwork/app/controllers/application_controller.rb` thanks for pointing me that out!

Reply:
See commit 064bc7025aab4445d78e4e6be821519a41e1ec42 : I've wrote a decorator for the Controller that serves the resources and sets the right CORS policy. Hope it's good now!